Add strict:true to tsconfig

I cannot use this library in my application without setting `strict:false` or `skipLibCheck:true` in my tsconfig becuase this library contains type errors. I suggest adding this option, then fix all of the errors reported by TypeScript compiler.

These are the errors when `strict` flag is set to `true`:
```
src/client.ts:45:10 - error TS2564: Property 'token' has no initializer and is not definitely assigned in the constructor.

45   public token: string
            ~~~~~

src/client.ts:47:10 - error TS2564: Property 'emit' has no initializer and is not definitely assigned in the constructor.

47   public emit: (event: PutioAPIClientEventTypes, ...args: any[]) => void
            ~~~~

src/client.ts:48:10 - error TS2416: Property 'once' in type 'PutioAPIClient' is not assignable to the same property in base type 'Emitter'.
  Type '(event: PutioAPIClientEventTypes, listener: EventListener) => void' is not assignable to type 'EmitterMethod'.
    Types of parameters 'event' and 'type' are incompatible.
      Type 'string' is not assignable to type 'PutioAPIClientEventTypes'.

48   public once: (
            ~~~~

src/client.ts:48:10 - error TS2564: Property 'once' has no initializer and is not definitely assigned in the constructor.

48   public once: (
            ~~~~

src/client.ts:52:10 - error TS2416: Property 'on' in type 'PutioAPIClient' is not assignable to the same property in base type 'Emitter'.
  Type '(event: PutioAPIClientEventTypes, listener: EventListener) => void' is not assignable to type 'EmitterMethod'.
    Types of parameters 'event' and 'type' are incompatible.
      Type 'string' is not assignable to type 'PutioAPIClientEventTypes'.

52   public on: (event: PutioAPIClientEventTypes, listener: EventListener) => void
            ~~

src/client.ts:52:10 - error TS2564: Property 'on' has no initializer and is not definitely assigned in the constructor.

52   public on: (event: PutioAPIClientEventTypes, listener: EventListener) => void
            ~~

src/client.ts:53:10 - error TS2416: Property 'off' in type 'PutioAPIClient' is not assignable to the same property in base type 'Emitter'.
  Type '(event: PutioAPIClientEventTypes, listener: EventListener) => void' is not assignable to type 'EmitterMethod'.
    Types of parameters 'event' and 'type' are incompatible.
      Type 'string' is not assignable to type 'PutioAPIClientEventTypes'.

53   public off: (event: PutioAPIClientEventTypes, listener: EventListener) => void
            ~~~

src/client.ts:53:10 - error TS2564: Property 'off' has no initializer and is not definitely assigned in the constructor.

53   public off: (event: PutioAPIClientEventTypes, listener: EventListener) => void
            ~~~

src/client.ts:57:10 - error TS2564: Property 'Account' has no initializer and is not definitely assigned in the constructor.

57   public Account: Account
            ~~~~~~~

src/resources/Auth.ts:49:41 - error TS2532: Object is possibly 'undefined'.

49       `/oauth2/authorizations/clients/${app.client_id}?client_secret=${app.client_secret}`,
                                           ~~~

src/resources/Auth.ts:49:72 - error TS2532: Object is possibly 'undefined'.

49       `/oauth2/authorizations/clients/${app.client_id}?client_secret=${app.client_secret}`,
                                                                          ~~~

src/resources/Payment/Payment.ts:67:21 - error TS7006: Parameter 'args' implicitly has an 'any' type.

67   public ChangePlan(args) {
                       ~~~~

src/resources/Payment/Payment.ts:113:28 - error TS7006: Parameter 'data' implicitly has an 'any' type.

113   public AddWaitingPayment(data) {
                               ~~~~

src/utils/index.ts:8:44 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(o: {}): string[]', gave the following error.
    Argument of type 'object | null' is not assignable to parameter of type '{}'.
      Type 'null' is not assignable to type '{}'.
  Overload 2 of 2, '(o: object): string[]', gave the following error.
    Argument of type 'object | null' is not assignable to parameter of type 'object'.
      Type 'null' is not assignable to type 'object'.

8   typeof input === 'object' && Object.keys(input).includes('error_type')
                                             ~~~~~



Found 14 errors.
```

--- a/src/client.ts
+++ b/src/client.ts
@@ -45,10 +45,6 @@ class PutioAPIClient {
   public token: string | undefined
 
   public emitter: Emitter
-  public emit (event: PutioAPIClientEventTypes, ...args: any[]) { this.emitter.emit(event, ...args) }
-  public once (event: PutioAPIClientEventTypes, listener: EventListener) { this.once(event, listener) }
-  public on (event: PutioAPIClientEventTypes, listener: EventListener) { this.emitter.on(event, listener) }
-  public off (event: PutioAPIClientEventTypes, listener: EventListener) { this.emitter.off(event, listener) }
 
   public http: AxiosInstance
 
@@ -91,6 +87,22 @@ class PutioAPIClient {
     this.User = new User(this)
     this.Zips = new Zips(this)
     this.IFTTT = new IFTTT(this)
+  }
+
+  public emit(event: PutioAPIClientEventTypes, ...args: any[]) {
+    this.emitter.emit(event, ...args)
+  }
+
+  public once(event: PutioAPIClientEventTypes, listener: EventListener) {
+    this.once(event, listener)
+  }
+
+  public on(event: PutioAPIClientEventTypes, listener: EventListener) {
+    this.emitter.on(event, listener)
+  }
+
+  public off(event: PutioAPIClientEventTypes, listener: EventListener) {
+    this.emitter.off(event, listener)
   }
 
   public setToken(token: string): PutioAPIClient {

--- a/src/client.ts
+++ b/src/client.ts
@@ -32,7 +32,7 @@ import {
   PutioAPIClientEventTypes,
 } from './types'
 
-class PutioAPIClient implements Emitter {
+class PutioAPIClient {
   public static EVENTS = PutioAPIClientEventTypes
 
   public static DEFAULT_OPTIONS: IPutioAPIClientOptions = {
@@ -42,19 +42,16 @@ class PutioAPIClient implements Emitter {
   }
 
   public options: IPutioAPIClientOptions
-  public token: string
+  public token: string | undefined
 
-  public emit: (event: PutioAPIClientEventTypes, ...args: any[]) => void
-  public once: (
-    event: PutioAPIClientEventTypes,
-    listener: EventListener,
-  ) => void
-  public on: (event: PutioAPIClientEventTypes, listener: EventListener) => void
-  public off: (event: PutioAPIClientEventTypes, listener: EventListener) => void
+  public emitter: Emitter
+  public emit (event: PutioAPIClientEventTypes, ...args: any[]) { this.emitter.emit(event, ...args) }
+  public once (event: PutioAPIClientEventTypes, listener: EventListener) { this.once(event, listener) }
+  public on (event: PutioAPIClientEventTypes, listener: EventListener) { this.emitter.on(event, listener) }
+  public off (event: PutioAPIClientEventTypes, listener: EventListener) { this.emitter.off(event, listener) }
 
   public http: AxiosInstance
 
-  public Account: Account
   public Auth: Auth
   public Config: Config
   public Events: Events
@@ -74,7 +71,7 @@ class PutioAPIClient implements Emitter {
   public Zips: Zips
 
   constructor(options: IPutioAPIClientOptions) {
-    EventEmitter(this)
+    this.emitter = EventEmitter()
     this.options = { ...PutioAPIClient.DEFAULT_OPTIONS, ...options }
     this.http = this.createHTTPClient()
     this.Auth = new Auth(this)

--- a/src/resources/Auth.ts
+++ b/src/resources/Auth.ts
@@ -40,7 +40,7 @@ export default class Auth {
   }: {
     username: string
     password: string
-    app?: {
+    app: {
       client_id: string
       client_secret: string
     }

--- a/src/resources/Payment/Payment.ts
+++ b/src/resources/Payment/Payment.ts
@@ -64,7 +64,7 @@ export default class Payment {
     })
   }
 
-  public ChangePlan(args) {
+  public ChangePlan(args: any) {
     return this.GetPlanChangeUrls(args)
   }
 
@@ -110,7 +110,7 @@ export default class Payment {
     })
   }
 
-  public AddWaitingPayment(data) {
+  public AddWaitingPayment(data: any) {
     return this.client.post('/payment/paddle_waiting_payment', { data })
   }
 }

--- a/src/utils/index.ts
+++ b/src/utils/index.ts
@@ -3,9 +3,9 @@ import { IPutioAPIClientError, IPutioAPIClientErrorData } from '../types'
 export const identity = <T>(arg: T) => arg
 
 export const isPutioAPIErrorResponse = (
-  input: unknown,
+  input: any,
 ): input is IPutioAPIClientErrorData =>
-  typeof input === 'object' && Object.keys(input).includes('error_type')
+  Object.keys(input).includes('error_type')
 
 export const isPutioAPIError = (
   input: unknown,

--- a/src/utils/index.ts
+++ b/src/utils/index.ts
@@ -3,13 +3,16 @@ import { IPutioAPIClientError, IPutioAPIClientErrorData } from '../types'
 export const identity = <T>(arg: T) => arg
 
 export const isPutioAPIErrorResponse = (
-  input: any,
+  input: unknown,
 ): input is IPutioAPIClientErrorData =>
-  Object.keys(input).includes('error_type')
+  typeof input === 'object' &&
+  !!input &&
+  !!(input as Record<string, unknown>).error_type
 
 export const isPutioAPIError = (
   input: unknown,
 ): input is IPutioAPIClientError =>
   typeof input === 'object' &&
+  !!input &&
   !!(input as Record<string, unknown>).data &&
   isPutioAPIErrorResponse((input as Record<string, unknown>).data)

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "strict": true,
     "moduleResolution": "node",
     "target": "es5",
     "module": "es6",

